### DISCUSSION TITLE
application security?

### DISCUSSION THREAD

**Discussion by trovador01** [immich-app/immich#2706] [General]
Hello good afternoon, I have been testing the application and it is very good. The only thing that worries me is is it safe? In other words, I see that when accessing the application outside the home network, it does not have a certificate and the information can only be accessed with a username and password. This is enough?

**Comment by samip5** [comment]
Hello,

People tend to host the app behind HTTPS and/or Cloudflare Tunnels.

**Reply by alextran1502** [reply]
The term you want to look into is called Reverse Proxy, using tool such as NGINX or Caddy

**Reply by trovador01** [reply]
I have it mounted in docker desktop windows, I have tried to mount NGINX in docker but I can't get it to mount. You can help

**Reply by simone-viozzi** [reply]
If you need an example with traefik, [here](https://github.com/simone-viozzi/my-server) is my configuration. You can find the immich docker compose file under immich, and traefik under traefik. I use let's encrypt to manage the certs.
The network `proxy` needs to be crated manually because it's shared between more than one docker-compose files. If you put immich and traefik in the same file, you will not need it.

**Comment by Pheggas** [comment]
You would need a reverse proxy. I will be doing some guides in upcoming days and pull it to official documentation so anyone can easily see it while reading the documentation
